src/com/mendix/widget/PullToRefresh/handlers/PullToRefresh.ts

```diff
@@ -0,0 +1,215 @@
+        const touch = event.touches[0];
+        const touchElement = document.elementFromPoint(touch.clientX, touch.clientY);
+        this.pullMoveY = event.touches[0].screenY;
+
+        if (!this.enable || this.state === "refreshing" || this.updating || this.state === "scrolling") return;
+
+        if (this.isScrollActive(touchElement as HTMLElement)) return;
+
+        if (this.state === "pending") {
+            domClass.add(pullToRefreshElement.id, `${classPrefix}pull`);
+            this.state = "pulling";
+            this.update();
+        }
+        if (this.pullStart.screenY && this.pullMoveY) {
+            this.distance = this.pullMoveY - this.pullStart.screenY;
+        }
+        if (this.distance > 0 && Math.abs(this.lastDistance - this.distance) > 1) {
+            this.lastDistance = this.distance;
+            event.preventDefault();
+            domStyle.set(pullToRefreshElement, cssProp, `${this.distanceResisted}px`);
+            this.distanceResisted = this.settings.resistanceFunction(this.distance / thresholdDistance)
+                * Math.min(maximumDistance, this.distance);
+
+            if (this.state === "pulling" && this.distanceResisted > thresholdDistance) {
+                domClass.add(pullToRefreshElement.id, `${classPrefix}release`);
+                this.state = "releasing";
+                this.update();
+            }
+
+            if (this.state === "releasing" && this.distanceResisted < thresholdDistance) {
+                domClass.remove(pullToRefreshElement.id, `${classPrefix}release`);
+                this.state = "pulling";
+                this.update();
+            }
+        }
+    }
+
+    private onTouchEnd() {
+        const { pullToRefreshElement, onRefresh, cssProp, classPrefix } = this.settings;
+
+        if (this.state === "releasing" && this.distanceResisted > this.settings.thresholdDistance) {
+            this.state = "refreshing";
+
+            domStyle.set(pullToRefreshElement, cssProp, `${this.settings.reloadDistance}px`);
+            domClass.add(pullToRefreshElement.id, `${classPrefix}refresh`);
+
+            this.timeout = setTimeout(() => {
+                const afterRefresh = onRefresh(this.onReset);
+                if (afterRefresh && typeof afterRefresh.then === "function") {
+                    afterRefresh.then(() => this.onReset());
+                }
+                if (!afterRefresh && !onRefresh.length) {
+                    this.onReset();
+                }
+            }, this.settings.refreshTimeout);
+        } else {
+            if (this.state === "refreshing") return;
+
+            domStyle.set(pullToRefreshElement, cssProp, "0px");
+            this.state = "pending";
+        }
+
+        this.update();
+        domClass.remove(pullToRefreshElement.id, `${classPrefix}release ${classPrefix}pull`);
+        this.pullStart.screenY = this.pullStart.screenX = this.pullMoveY = 0;
```

Comment:
Why reset, this is already done in the start?
`this.pullStart = { screenX: 0, screenY: 0 };` looks cleaner